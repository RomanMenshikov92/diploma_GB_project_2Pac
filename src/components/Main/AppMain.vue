<template lang="pug">
main.main
  .container.main__container
    router-view
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  name: "AppMain",
});
</script>

<style lang="sass"></style>
