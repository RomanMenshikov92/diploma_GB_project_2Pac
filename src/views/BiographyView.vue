<template lang="pug">
.biography
  SectionDevelopment(class="biography__development" :dataTitle="'titleBiography'")
</template>

<script lang="ts">
import { defineComponent, onMounted } from "vue";
import SectionDevelopment from "@/components/Common/SectionDevelopment.vue";
import { useTitleStore } from "@/stores/Common/useTitleStore";

export default defineComponent({
  name: "BiographyView",
  components: {
    SectionDevelopment,
  },
  setup() {
    // use store
    const titleStore = useTitleStore();

    // store state
    const titleItem = titleStore.getTitleItems;

    onMounted(() => {
      console.log("onMounted page biography");
    });
    return {
      titleItem,
    };
  },
});
</script>

<style lang="sass"></style>
