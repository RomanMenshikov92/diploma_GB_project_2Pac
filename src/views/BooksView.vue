<template lang="pug">
.books
  SectionDevelopment(class="books__development" :dataTitle="'titleBooks'")
</template>

<script lang="ts">
import { defineComponent, onMounted } from "vue";
import SectionDevelopment from "@/components/Common/SectionDevelopment.vue";
import { useTitleStore } from "@/stores/Common/useTitleStore";

export default defineComponent({
  name: "BooksView",
  components: {
    SectionDevelopment,
  },
  setup() {
    // use store
    const titleStore = useTitleStore();

    // store state
    const titleItem = titleStore.getTitleItems;

    onMounted(() => {
      console.log("onMounted page books");
    });
    return {
      titleItem,
    };
  },
});
</script>

<style lang="sass"></style>
