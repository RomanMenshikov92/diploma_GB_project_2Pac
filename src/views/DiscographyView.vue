<template lang="pug">
.discography
  SectionDevelopment(class="discography__development" :dataTitle="'titleDiscography'")
</template>

<script lang="ts">
import { defineComponent, onMounted } from "vue";
import SectionDevelopment from "@/components/Common/SectionDevelopment.vue";
import { useTitleStore } from "@/stores/Common/useTitleStore";

export default defineComponent({
  name: "DiscographyView",
  components: {
    SectionDevelopment,
  },
  setup() {
    // use store
    const titleStore = useTitleStore();

    // store state
    const titleItem = titleStore.getTitleItems;

    onMounted(() => {
      console.log("onMounted page discography");
    });
    return {
      titleItem,
    };
  },
});
</script>

<style lang="sass"></style>
