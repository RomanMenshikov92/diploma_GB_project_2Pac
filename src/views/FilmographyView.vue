<template lang="pug">
.filmography
  SectionDevelopment(class="filmography__development" :dataTitle="'titleFilmography'")
</template>

<script lang="ts">
import { defineComponent, onMounted } from "vue";
import SectionDevelopment from "@/components/Common/SectionDevelopment.vue";
import { useTitleStore } from "@/stores/Common/useTitleStore";

export default defineComponent({
  name: "FilmographyView",
  components: {
    SectionDevelopment,
  },
  setup() {
    // use store
    const titleStore = useTitleStore();

    // store state
    const titleItem = titleStore.getTitleItems;

    onMounted(() => {
      console.log("onMounted page filmography");
    });
    return {
      titleItem,
    };
  },
});
</script>

<style lang="sass"></style>
