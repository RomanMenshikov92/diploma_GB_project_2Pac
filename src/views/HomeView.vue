<template lang="pug">
.home
  SectionHome.home__main
  SectionBiography.home__biography
  SectionDiscography.home__discography
  SectionVideography.home__videography
  SectionFilmography.home__filmography
  SectionBooks.home__books
</template>

<script lang="ts">
import { defineComponent, onBeforeUnmount } from "vue";
import { useHomeVideoStore } from "@/stores/Home/useHomeVideoStore";
import SectionHome from "@/components/Main/Section/Home/SectionHome.vue";
import SectionBiography from "@/components/Main/Section/Biography/SectionBiography.vue";
import SectionDiscography from "@/components/Main/Section/Discography/SectionDiscography.vue";
import SectionVideography from "@/components/Main/Section/Videography/SectionVideography.vue";
import SectionFilmography from "@/components/Main/Section/Filmography/SectionFilmography.vue";
import SectionBooks from "@/components/Main/Section/Books/SectionBooks.vue";

export default defineComponent({
  name: "HomeView",
  components: {
    SectionHome,
    SectionBiography,
    SectionDiscography,
    SectionVideography,
    SectionFilmography,
    SectionBooks,
  },
  setup() {
    // use store
    const homeVideoStore = useHomeVideoStore();

    // reset
    onBeforeUnmount(() => {
      homeVideoStore.reset();
    });

    // return
    return {
      onBeforeUnmount,
    };
  },
});
</script>

<style lang="sass" scoped></style>
