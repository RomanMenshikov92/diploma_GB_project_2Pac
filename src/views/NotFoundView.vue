<template lang="pug">
.not-found
  .not-found__text
    h2.title.title-reset.not-found__title
      NotFoundIcon
    p.paragraph.paragraph-reset.not-found__paragraph Запрашиваемая страница не найдена
    RouterLink.not-found__link(to="/") На главную
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { RouterLink } from "vue-router";
import NotFoundIcon from "@/components/Icon/NotFoundIcon.vue";

export default defineComponent({
  name: "NotFoundView",
  components: {
    RouterLink,
    NotFoundIcon,
  },
  setup() {
    return {};
  },
});
</script>

<style lang="sass" scoped></style>
