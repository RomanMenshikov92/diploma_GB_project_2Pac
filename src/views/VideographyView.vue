<template lang="pug">
.videography
  SectionDevelopment(class="videography__development" :dataTitle="'titleVideography'")
</template>

<script lang="ts">
import { defineComponent, onMounted } from "vue";
import SectionDevelopment from "@/components/Common/SectionDevelopment.vue";
import { useTitleStore } from "@/stores/Common/useTitleStore";

export default defineComponent({
  name: "VideographyView",
  components: {
    SectionDevelopment,
  },
  setup() {
    // use store
    const titleStore = useTitleStore();

    // store state
    const titleItem = titleStore.getTitleItems;

    onMounted(() => {
      console.log("onMounted page videography");
    });
    return {
      titleItem,
    };
  },
});
</script>

<style lang="sass"></style>
