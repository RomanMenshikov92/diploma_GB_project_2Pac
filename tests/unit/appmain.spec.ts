import { mount } from "@vue/test-utils";
import AppMain from "@/components/Main/AppMain.vue";
import HomeView from "@/views/HomeView.vue";
import BiographyView from "@/views/BiographyView.vue";
import DiscographyView from "@/views/DiscographyView.vue";
import VideographyView from "@/views/VideographyView.vue";
import FilmographyView from "@/views/FilmographyView.vue";
import BooksView from "@/views/BooksView.vue";
import NotFoundView from "@/views/NotFoundView.vue";
import { Pinia, createPinia } from "pinia";
import { createRouter, createWebHistory } from "vue-router";

jest.mock("@/views/HomeView.vue", () => ({
  name: "HomeView",
  render: jest.fn(),
}));

jest.mock("@/views/BiographyView.vue", () => ({
  name: "BiographyView",
  render: jest.fn(),
}));

jest.mock("@/views/DiscographyView.vue", () => ({
  name: "DiscographyView",
  render: jest.fn(),
}));

jest.mock("@/views/VideographyView.vue", () => ({
  name: "VideographyView",
  render: jest.fn(),
}));

jest.mock("@/views/FilmographyView.vue", () => ({
  name: "FilmographyView",
  render: jest.fn(),
}));

jest.mock("@/views/BooksView.vue", () => ({
  name: "BooksView",
  render: jest.fn(),
}));

jest.mock("@/views/NotFoundView.vue", () => ({
  name: "NotFoundView",
  render: jest.fn(),
}));

describe("AppMain.vue", () => {
  it("renders correct Pug in main layout", () => {
    const router = createRouter({
      history: createWebHistory(),
      routes: [
        {
          path: "/",
          name: "home",
          component: HomeView,
        },
        {
          path: "/biography",
          name: "biography",
          component: BiographyView,
        },
        {
          path: "/discography",
          name: "discography",
          component: DiscographyView,
        },
        {
          path: "/videography",
          name: "videography",
          component: VideographyView,
        },
        {
          path: "/filmography",
          name: "filmography",
          component: FilmographyView,
        },
        {
          path: "/books",
          name: "books",
          component: BooksView,
        },
        {
          path: "/:pathMatch(.*)",
          name: "not-found",
          component: NotFoundView,
        },
      ],
    });

    const pinia: Pinia = createPinia();

    const wrapper = mount(AppMain, {
      global: {
        plugins: [pinia, router],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
  it("renders correct components for each route", async () => {
    const router = createRouter({
      history: createWebHistory(),
      routes: [
        {
          path: "/",
          name: "home",
          component: HomeView,
        },
        {
          path: "/biography",
          name: "biography",
          component: BiographyView,
        },
        {
          path: "/discography",
          name: "discography",
          component: DiscographyView,
        },
        {
          path: "/videography",
          name: "videography",
          component: VideographyView,
        },
        {
          path: "/filmography",
          name: "filmography",
          component: FilmographyView,
        },
        {
          path: "/books",
          name: "books",
          component: BooksView,
        },
        {
          path: "/:pathMatch(.*)",
          name: "not-found",
          component: NotFoundView,
        },
      ],
    });

    const wrapper = mount(AppMain, {
      global: {
        plugins: [router],
      },
    });

    await router.isReady();
    await router.push("/");
    expect(wrapper.findComponent({ name: "HomeView" }).exists()).toBe(true);
    await router.push("/biography");
    expect(wrapper.findComponent({ name: "BiographyView" }).exists()).toBe(true);
    await router.push("/discography");
    expect(wrapper.findComponent({ name: "DiscographyView" }).exists()).toBe(true);
    await router.push("/videography");
    expect(wrapper.findComponent({ name: "VideographyView" }).exists()).toBe(true);
    await router.push("/filmography");
    expect(wrapper.findComponent({ name: "FilmographyView" }).exists()).toBe(true);
    await router.push("/books");
    expect(wrapper.findComponent({ name: "BooksView" }).exists()).toBe(true);
    await router.push("/:pathMatch(.*)");
    expect(wrapper.findComponent({ name: "NotFoundView" }).exists()).toBe(true);
  });
});
