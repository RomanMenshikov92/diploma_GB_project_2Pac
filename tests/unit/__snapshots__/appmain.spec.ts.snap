// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMain.vue renders correct Pug in main layout 1`] = `
<main class="main">
  <div class="container main__container">
    <!---->
  </div>
</main>
`;
